pkgs/tools/networking/zap/default.nix

```
@@ -1,46 +1,82 @@
-{ lib, stdenv, fetchurl, jre, runtimeShell }:
+{ lib, stdenv, fetchurl, jre, undmg, runtimeShell }:
 
-stdenv.mkDerivation rec {
+let
   pname = "zap";
   version = "2.15.0";
-  src = fetchurl {
-    url = "https://github.com/zaproxy/zaproxy/releases/download/v${version}/ZAP_${version}_Linux.tar.gz";
-    sha256 = "sha256-ZBDhlrqrRYqSBOKar7V0X8oAOipsA4byxuXAS2diH6c=";
+  sources = {
+    linux = fetchurl {
+      url =
+        "https://github.com/zaproxy/zaproxy/releases/download/v${version}/ZAP_${version}_Linux.tar.gz";
+      sha256 = "sha256-ZBDhlrqrRYqSBOKar7V0X8oAOipsA4byxuXAS2diH6c=";
+    };
+    darwin-x86_64 = fetchurl {
+      url =
+        "https://github.com/zaproxy/zaproxy/releases/download/v${version}/ZAP_${version}.dmg";
+      sha256 = "sha256-rgJUA+Rs3v/wE80MO4jY7cWhg6dtqmPLYsfGKQBTN6U=";
+    };
+    darwin-aarch64 = fetchurl {
```

Use of common platform values also available in `stdenv.system` as keys in `sources` can simplify setting `src` below:
```suggestion
  sources = rec {
    x86_64-linux = fetchurl {
      url =
        "https://github.com/zaproxy/zaproxy/releases/download/v${version}/ZAP_${version}_Linux.tar.gz";
      sha256 = "sha256-ZBDhlrqrRYqSBOKar7V0X8oAOipsA4byxuXAS2diH6c=";
    };
    aarch64-linux = x86_64-linux;
    x86_64-darwin = fetchurl {
      url =
        "https://github.com/zaproxy/zaproxy/releases/download/v${version}/ZAP_${version}.dmg";
      sha256 = "sha256-rgJUA+Rs3v/wE80MO4jY7cWhg6dtqmPLYsfGKQBTN6U=";
    };
    aarch64-darwin = fetchurl {
```